sources/core/Xenko.Core.Mathematics/Matrix.cs

```diff
@@ -707,11 +707,98 @@ public bool Decompose(out Vector3 scale, out Quaternion rotation, out Vector3 tr
             return true;
         }
 
+        /// <summary>
+        /// Decomposes a matrix into a rotation.
+        /// </summary>
+        /// <param name="matrix">When the method completes, contains the rotation component of the decomposed matrix.</param>
+        /// <remarks>
+        /// This method is designed to decompose an SRT transformation matrix only.
+        /// </remarks>
+        public bool GetRotationMatrix(out Matrix matrix)
```

Comment:
We have `Matrix.Decompose`.
I suppose the reason to add one that deals with rotation or scaling only is to avoid computing the other parts?

If yes, maybe we should at least rename it in a similar way (i.e. `DecomposeRotation` and `DecomposeScale`)?

Reply:
Yes, if we only need a rotation or scale, I don't want to compute the other parts.

Renaming them to Decompose(type) isn't a bad idea, although "decompose" isn't an intuitive as "get", in my humble opinion... but consistency is more important.